$MirOS: ports/print/cups/patches/patch-scheduler_cups-lpd_c,v 1.1.7.1 2005/03/18 15:51:00 tg Exp $
--- scheduler/cups-lpd.c.orig	Mon Apr 24 13:49:36 2006
+++ scheduler/cups-lpd.c	Tue May  9 17:39:40 2006
@@ -1367,7 +1367,7 @@ remove_jobs(const char *dest,		/* I - De
 
     request = ippNewRequest(IPP_CANCEL_JOB);
 
-    sprintf(uri, "ipp://localhost/jobs/%d", id);
+    snprintf(uri, sizeof(uri), "ipp://localhost/jobs/%d", id);
     ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_URI, "job-uri", NULL, uri);
 
     ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_NAME,
